### PR TITLE
[fx/DDP] add nested ctx_manager test for DDP Dynamo

### PR DESCRIPTION
**Pull request by jon-chuang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114056

### CODE REVIEW

File: test/distributed/test_c10d_nccl.py [hunk 1]
```diff
@@ -2018,17 +2018,6 @@ def _test_grad_layout(self, replica_devices, layer_devs, local_batch_size):
         local_batch_start = self.rank * local_batch_size
         local_batch_end = (self.rank + 1) * local_batch_size
 
-        # Reducer.cpp sneakily creates one "initial bucket" that ignores the "bucket_cap_mb"
-        # argument.  The following makes sure the initial bucket also complies.
-        @contextmanager
```
Comment (jon-chuang) [inline test/distributed/test_c10d_nccl.py:2023]: We just move this into common_distributed

**Review by wconstab — APPROVED** [review]

**Comment by jon-chuang** [comment]
Currently blocked on https://github.com/pytorch/pytorch/issues/114338

**Comment by jon-chuang** [comment]
Will be fixed by: https://github.com/pytorch/pytorch/pull/114343

**Pull request closed without merging**